lib/importeur/active_record_postgres_loader.rb

```diff
@@ -44,20 +44,24 @@ def store_record(attrs, records)
     def records_for_batch(batch_ids)
       model
         .with_deleted
-        .where(primary_key => batch_ids)
+        .joins(batch_lookup_join_sql(batch_ids))
```

Comment:
Don't we need to use inner join here, or add something like 

````
where('imported.primary_key IS NOT NULL')
````

(and also rename alias to be something else than imported)

It may still be working because of index_by, but seems the outer join will make it return rows for all records in db.

Reply:
You are right. I just quickly did it before leaving. We need the outer join when deleting, because we delete the ones that are not in the intersecting set. Here we want only those.